graph/patterns/longRunningOperations.md

```diff
@@ -0,0 +1,192 @@
+
+## When to use this pattern
+
+Any API call that is expected to take longer than 1 second in the 99th percentile should use the long running operations pattern.
+
+How do you select which flavor of LRO pattern to use? An API designer can follow these
+heuristics:
+
+1.  If a service can create a resource with a minimal latency and continue
+    updating its status according to the well-defined and stable state
+    transition model until completion, then the RELO model is the best choice.
+
+2.  Otherwise, a service should follow the Stepwise Operation pattern. 
+ 
+
+## Issues and considerations
+
+- One or more clients MUST be able to monitor and operate on the same resource at the same time.
+
+- The state of the system SHOULD always be discoverable and testable. Clients
+    SHOULD be able to determine the system state even if the operation tracking
+    resource is no longer active. Clients MAY issue a GET on some resource to
+    determine the state of a long running operation.
+
+-  The long running operations pattern SHOULD work for clients looking to "fire and forget"
+    and for clients looking to actively monitor and act upon results.
+
+-  The long running operations pattern might be supplemented by the [change notification
+    pattern](change-notification.md).
+
+-  Cancellation of a long running operation does not explicitly mean a rollback. On a per API-defined case, it
+    might mean a rollback, or compensation, or completion, or partial completion,
+    etc. Following a canceled operation, the API should return a consistent state that allows
+    continued service.
+
+-  A recommended minimum retention time for a stepwise operation is 24 hours.
+    Operations SHOULD transition to "tombstone" for an additional period of time
+    prior to being purged from the system.
+    
+-  Services that provides a new operation resource MUST support GET semantics on the operation.
+
+
+
+## Examples
+
+
+###  Create a new resource using RELO
+
+A client wants to provision a new database
+
+```
+POST https://graph.microsoft.com/v1.0/databases/
+
+{
+"id": "db1",
+}
+```
+
+The API responds synchronously that the database has been created and indicates
+that the provisioning operation is not fully completed by including the
+Operation-Location header and status property in the response payload.
+
+```
+HTTP/1.1 201 Created
+Content-Location: https://graph.microsoft.com/v1.0/databases/db1
+
+{
+"id": "db1",
+"Status": "Provisioning",
+[ … other fields for "database" …]
+}
+```
+
+### Create a new resource using the Stepwise Operation
+
+```
+POST https://graph.microsoft.com/v1.0/databases/
+
+{
+"id": "db1",
+}
+```
+
+The API responds synchronously that the request has been accepted and includes
+the Location header with an operation resource for further polling.
+
+```
+HTTP/1.1 202 Accepted
+
+Location: https://graph.microsoft.com/v1.0/operations/123
```

Comment:
We may want to avoid the creation of a root level operations resource since we don't have a way to do this for multiple workloads in Graph

```suggestion
Location: https://graph.microsoft.com/v1.0/databases/operations/123
```